test/test_local_io.py

```diff
@@ -655,6 +655,42 @@ def filepath_fn(name: str) -> str:
         # __len__ Test: returns the length of source DataPipe
         self.assertEqual(3, len(saver_dp))
 
+    @skipIfNoIoPath
+    def test_io_path_saver_file_lock(self):
+        def filepath_fn(name: str) -> str:
+            return os.path.join(self.temp_dir.name, os.path.basename(name))
+
+        # Same filename with different name
+        name_to_data = {"1.txt": b"DATA1", "1.txt": b"DATA2", "2.txt": b"DATA3", "2.txt": b"DATA4"}  # noqa: F601
+
+        # Add sharding_filter to shard data into 2
+        source_dp = IterableWrapper(list(name_to_data.items())).sharding_filter()
+
+        # Use appending as the mode
+        saver_dp = source_dp.save_by_iopath(filepath_fn=filepath_fn, mode="ab")
+
+        import torch.utils.data.graph_settings
+
+        def init_fn(worker_id):
+            info = torch.utils.data.get_worker_info()
+            num_workers = info.num_workers
+            datapipe = info.dataset
+            torch.utils.data.graph_settings.apply_sharding(datapipe, num_workers, worker_id)
+
+        from torch.utils.data import DataLoader
+
+        num_workers = 2
+        line_lengths = []
+        dl = DataLoader(saver_dp, num_workers=num_workers, worker_init_fn=init_fn)
```

Comment:
Could you please add multiprocessing context here?
```suggestion
        dl = DataLoader(saver_dp, num_workers=num_workers, worker_init_fn=init_fn, multiprocessing_context="spawn")
```